File: api-flow-builder/view/Theme.js
export class Theme {
    static THEME = {
        LIGHT: {
            bg: '#F6F7FB',
            panelBg: '#FFFFFF',
            cardBg: '#FFFFFF',
            sectionBg: '#F9FAFC',
            border: '#E2E6ED',
            text: '#1F2933',
            muted: '#6B7280',
            mutedDim: '#9AA5B1',

            primary: '#FF6C37',
            headerGradient: 'linear-gradient(135deg,#FF6C37,#FF8F5B)',
            headerText: '#FFFFFF',
            templateText: '#2563EB',
            templateChipBg: 'rgba(37,99,235,0.12)',
            templateChipBorder: 'rgba(37,99,235,0.28)',
            templateChipText: '#1D4ED8',
            tooltipBg: '#0F172A',
            tooltipText: '#F8FAFC',
            tooltipBorder: 'rgba(15,23,42,0.35)',

            inputBg: '#FFFFFF',
            navBg: '#FFFFFF',

            jsonBg: '#F7F8FB',
            jsonText: '#111827',
            jsonBorder: '#E2E6ED',

            addBtnBg: '#FFF3EE',
            addBtnBorder: '#FF6C37',

            stepNumBg: '#FFF3EE',

            sectionShadow: '0 1px 3px rgba(15,23,42,0.08)',
            cardShadow: '0 1px 2px rgba(15,23,42,0.06)',
            navShadow: '0 2px 20px rgba(15,23,42,0.12)',

            cardHdrBg: '#F9FAFC',
            cardHdrBorder: '#E2E6ED',

            emptyIcon: '#C7CED8',
            emptyText: '#6B7280',
            emptySubText: '#9AA5B1',

            rmColor: '#9AA5B1',
            rmHoverBg: 'rgba(239,68,68,0.12)',

            addHoverBg: 'rgba(255,108,55,0.18)',

            focusRing: '0 0 0 3px rgba(255,108,55,0.28)',

            toggleBg: '#F1F3F8',
            navHoverBg: 'rgba(255,255,255,0.2)',
            danger: '#E44D3A',
            statusLiveBg: '#E8F5EC',
            statusLiveText: '#157347',
            statusDirtyBg: '#FFF4E5',
            statusDirtyText: '#B45309',
            statusErrorBg: '#FDECEC',
            statusErrorText: '#B91C1C',
            methodBadge: {
                GET: { bg: '#E8F5EC', color: '#157347' },
                POST: { bg: '#E7F0FF', color: '#1E40AF' },
                PUT: { bg: '#FFF4E5', color: '#B45309' },
                PATCH: { bg: '#F3E8FF', color: '#6D28D9' },
                DELETE: { bg: '#FDECEC', color: '#B91C1C' },
            },
            icon: {
                app: '▣ API',
                meta: '▤ Meta',
                environment: '◍ Environment',
                inputs: '⌗ Inputs',
                steps: '▸ Steps',
                emptySteps: '▫ Step',

                request: '⟶ Request',
                extract: '⤓ Extract',
                checks: '✔ Checks',

                remove: '✖',
                copied: '✔ Copied',

                statusLive: '● Live',
                statusDirty: '◔ Syncing',
                statusError: '⚠ Invalid JSON',

                previewTitle: '▦ JSON Output',
                previewSubtitle: '✎ Edit JSON to sync with form',

                toggleLight: '☀ Light',
                toggleDark: '☽ Dark',

                productTag: '◆ API Flow JSON Generator',
                appSubtitle: '⚙ Design - Preview - Export',

                previewPanel: '▧ Preview'
            },
            toggleIcon: 'L',
        },

        DARK: {
            bg: '#1B1C20',
            panelBg: '#1F2127',
            cardBg: '#23262D',
            sectionBg: '#20232A',
            border: '#2E323B',

            text: '#E6E8EE',
            muted: '#A0A7B4',
            mutedDim: '#7C8594',

            primary: '#FF6C37',
            headerGradient: 'linear-gradient(135deg,#FF6C37,#FF8F5B)',
            headerText: '#FFFFFF',
            templateText: '#60A5FA',
            templateChipBg: 'rgba(96,165,250,0.16)',
            templateChipBorder: 'rgba(96,165,250,0.35)',
            templateChipText: '#93C5FD',
            tooltipBg: '#0B1220',
            tooltipText: '#E2E8F0',
            tooltipBorder: 'rgba(148,163,184,0.25)',

            inputBg: '#1B1C20',
            navBg: '#1F2127',

            jsonBg: '#1B1C20',
            jsonText: '#E6E8EE',
            jsonBorder: '#2E323B',

            addBtnBg: 'rgba(255,108,55,0.12)',
            addBtnBorder: '#FF6C37',

            stepNumBg: 'rgba(255,108,55,0.18)',

            sectionShadow: '0 2px 10px rgba(0,0,0,0.45)',
            cardShadow: '0 2px 6px rgba(0,0,0,0.5)',
            navShadow: '0 2px 20px rgba(0,0,0,0.5)',

            cardHdrBg: '#20232A',
            cardHdrBorder: '#2E323B',

            emptyIcon: '#6E7685',
            emptyText: '#A0A7B4',
            emptySubText: '#7C8594',

            rmColor: '#A0A7B4',
            rmHoverBg: 'rgba(239,68,68,0.22)',

            addHoverBg: 'rgba(255,108,55,0.22)',

            focusRing: '0 0 0 3px rgba(255,108,55,0.28)',

            toggleBg: '#23262D',
            navHoverBg: 'rgba(255,255,255,0.2)',
            danger: '#F06A55',
            statusLiveBg: 'rgba(22,163,74,0.15)',
            statusLiveText: '#6EE7B7',
            statusDirtyBg: 'rgba(245,158,11,0.18)',
            statusDirtyText: '#FCD34D',
            statusErrorBg: 'rgba(239,68,68,0.18)',
            statusErrorText: '#FCA5A5',
            methodBadge: {
                GET: { bg: 'rgba(22,163,74,0.18)', color: '#6EE7B7' },
                POST: { bg: 'rgba(59,130,246,0.18)', color: '#93C5FD' },
                PUT: { bg: 'rgba(245,158,11,0.18)', color: '#FCD34D' },
                PATCH: { bg: 'rgba(139,92,246,0.18)', color: '#C4B5FD' },
                DELETE: { bg: 'rgba(239,68,68,0.18)', color: '#FCA5A5' },
            },
            icon: {
                app: '▣ API',
                meta: '▤ Meta',
                environment: '◍ Environment',
                inputs: '⌗ Inputs',
                steps: '▸ Steps',
                emptySteps: '▫ Step',

                request: '⟶ Request',
                extract: '⤓ Extract',
                checks: '✔ Checks',

                remove: '✖',
                copied: '✔ Copied',

                statusLive: '● Live',
                statusDirty: '◔ Syncing',
                statusError: '⚠ Invalid JSON',

                previewTitle: '▦ JSON Output',
                previewSubtitle: '✎ Edit JSON to sync with form',

                toggleLight: '☀ Light',
                toggleDark: '☽ Dark',

                productTag: '◆ API Flow JSON Generator',
                appSubtitle: '⚙ Design - Preview - Export',

                previewPanel: '▧ Preview'
            },
            toggleIcon: 'D',
        },
    };

    static isDark = false;
    static token = Theme.THEME.LIGHT;

    static spacing = {
        // Gap scale
        gap2:   '1px',
        gap5:   '3px',
        gap6:   '4px',
        gap8:   '5px',
        gap10:  '6px',
        gap12:  '7px',
        gap14:  '8px',
        gap19:  '10px',

        // Padding presets
        padBadge:       '1px 5px',
        padToggleBtn:   '2px 7px',
        padBtn:         '3px 8px',
        padAddBtn:      '3px 8px',
        padInput:       '5px 8px',
        padSelect:      '5px 24px 5px 8px',
        padCardHdr:     '6px 8px',
        padSectionHdr:  '8px 10px',
        padCardBody:    '8px',
        padSectionBody: '10px',
        padFormInner:   '10px',
        padMainHeader:  '6px 8px',
        padNavbar:      '0 14px',
        padPill:        '2px 5px',
        padCopyBtn:     '3px 10px',
        padJson:        '12px',
        padPreviewHdr:  '8px 12px',
        padEmpty:       '22px 12px',

        // Margin
        marginBottomLabel: '6px',
    };

    static style = {
        // Layout
        flex1: '1',
        flexNone: 'none',
        minWidth0: '0',
        minHeight0: '0',
        flexShrink0: '0',
        widthFull: '100%',
        heightFull: '100%',
        heightViewport: '100vh',
        width42Percent: '42%',
        minWidth360: '360px',
        margin0: '0',

        // Positioning
        positionRelative: 'relative',
        positionAbsolute: 'absolute',
        cursorColResize: 'col-resize',
        zIndex10: '10',
        leftHandle: '-3px',
        top0: '0',
        bottom0: '0',
        widthHandle: '6px',

        // Sizing
        height1px: '1px',
        heightNavbar: '54px',
        height22: '22px',
        width22: '22px',
        height26: '26px',
        width26: '26px',
        maxWidth180: '180px',
        maxWidth260: '260px',
        minHeight120: '120px',

        // Typography
        fontFamilyMono: 'monospace',
        fontFamilyBase: '"Inter", system-ui, sans-serif',
        fontFamilyCode: '"Fira Code","Cascadia Code","JetBrains Mono",monospace',
        fontSize68: '0.6rem',
        fontSize70: '0.62rem',
        fontSize72: '0.64rem',
        fontSize73: '0.65rem',
        fontSize75: '0.67rem',
        fontSize78: '0.69rem',
        fontSize82: '0.72rem',
        fontSize85: '0.75rem',
        fontSize87: '0.77rem',
        fontSize88: '0.79rem',
        fontSize95: '0.85rem',
        fontSize100: '0.9rem',
        fontSize120: '1.0rem',
        fontSize140: '1.2rem',
        fontWeight500: '500',
        fontWeight600: '600',
        fontWeight700: '700',
        letterSpacingTight: '-0.2px',
        letterSpacingMedium: '0.5px',
        letterSpacingWide: '0.8px',
        textUppercase: 'uppercase',
        lineHeightJson: '1.7',
        textOverflowEllipsis: 'ellipsis',

        // Borders
        borderRadius5: '5px',
        borderRadius6: '6px',
        borderRadius8: '8px',
        borderRadius10: '10px',
        borderRadius12: '12px',
        borderRadiusPill: '99px',
        borderRadiusCircle: '50%',

        // Effects
        opacity50: '0.5',
        opacity88: '0.88',
        opacity100: '1',
        transformUp1: 'translateY(-1px)',
        transform0: 'translateY(0)',

        // Backgrounds
        backgroundTransparent: 'transparent',
        backgroundNoRepeat: 'no-repeat',
        backgroundPosRight: 'right 10px center',
    };

    isDark;
    token;

    constructor(isDark = false) {
        this.isDark = isDark;
        this.token = isDark ? Theme.THEME.DARK : Theme.THEME.LIGHT;
    }

    static setMode(isDark = false) {
        Theme.isDark = isDark;
        Theme.token = isDark ? Theme.THEME.DARK : Theme.THEME.LIGHT;
    }
}

File: gwt-api-flow-builder/src/main/java/com/visit/gwtapiflowbuilder/client/theme/Theme.java
package com.visit.gwtapiflowbuilder.client.theme;

public final class Theme {
    public static final String FONT_BASE = "\"Inter\", system-ui, sans-serif";
    public static final String FONT_MONO = "\"Fira Code\",\"Cascadia Code\",\"JetBrains Mono\",monospace";
    public static final String TOGGLE_LABEL_LIGHT = "☀ Light";
    public static final String TOGGLE_LABEL_DARK = "☽ Dark";
    public static final String FONT_SIZE_68 = "10px";
    public static final String FONT_SIZE_72 = "10.5px";
    public static final String FONT_SIZE_78 = "11px";
    public static final String FONT_SIZE_82 = "11.5px";
    public static final String FONT_SIZE_87 = "12px";
    public static final String FONT_SIZE_95 = "13px";
    public static final String FONT_SIZE_140 = "18px";
    public static final String FONT_WEIGHT_500 = "500";
    public static final String FONT_WEIGHT_600 = "600";
    public static final String FONT_WEIGHT_700 = "700";
    public static final String LETTER_SPACING_TIGHT = "-0.2px";
    public static final String LETTER_SPACING_WIDE = "0.8px";
    public static final String GAP_6 = "6px";
    public static final String GAP_8 = "8px";
    public static final String GAP_10 = "10px";
    public static final String PAD_SECTION_HDR = "8px 10px";
    public static final String PAD_SECTION_BODY = "10px";
    public static final String PAD_FORM_INNER = "10px";
    public static final String PAD_INPUT = "5px 8px";
    public static final String PAD_NAVBAR = "0 14px";
    public static final String PAD_PREVIEW_HDR = "8px 12px";
    public static final String PAD_JSON = "12px";
    public static final String HEIGHT_NAVBAR = "54px";
    private static final Palette LIGHT = new Palette(
            "#F6F7FB",
            "#FFFFFF",
            "#F9FAFC",
            "#E2E6ED",
            "#1F2933",
            "#6B7280",
            "#9AA5B1",
            "#FF6C37",
            "#FFF3EE",
            "#F7F8FB",
            "#111827",
            "#E8F5EC",
            "#157347",
            "#FDECEC",
            "#B91C1C",
            "#E44D3A",
            "#2563EB",
            "linear-gradient(135deg,#FF6C37,#FF8F5B)"
    );
    private static final Palette DARK = new Palette(
            "#1B1C20",
            "#1F2127",
            "#20232A",
            "#2E323B",
            "#E6E8EE",
            "#A0A7B4",
            "#7C8594",
            "#FF6C37",
            "rgba(255,108,55,0.12)",
            "#1B1C20",
            "#E6E8EE",
            "rgba(22,163,74,0.15)",
            "#6EE7B7",
            "rgba(239,68,68,0.18)",
            "#FCA5A5",
            "#F06A55",
            "#60A5FA",
            "linear-gradient(135deg,#FF6C37,#FF8F5B)"
    );
    public static boolean IS_DARK = false;
    public static String COLOR_BG = "#F6F7FB";
    public static String COLOR_PANEL = "#FFFFFF";
    public static String COLOR_SECTION = "#F9FAFC";
    public static String COLOR_BORDER = "#E2E6ED";
    public static String COLOR_TEXT = "#1F2933";
    public static String COLOR_MUTED = "#6B7280";
    public static String COLOR_MUTED_DIM = "#9AA5B1";
    public static String COLOR_PRIMARY = "#FF6C37";
    public static String COLOR_PRIMARY_LIGHT = "#FFF3EE";
    public static String COLOR_JSON_BG = "#F7F8FB";
    public static String COLOR_JSON_TEXT = "#111827";
    public static String COLOR_STATUS_BG = "#E8F5EC";
    public static String COLOR_STATUS_TEXT = "#157347";
    public static String COLOR_STATUS_ERROR_BG = "#FDECEC";
    public static String COLOR_STATUS_ERROR_TEXT = "#B91C1C";
    public static String COLOR_DANGER = "#E44D3A";
    public static String COLOR_TEMPLATE = "#2563EB";
    public static String GRADIENT_HEADER = "linear-gradient(135deg,#FF6C37,#FF8F5B)";

    private Theme() {
    }

    public static void setMode(boolean dark) {
        IS_DARK = dark;
        apply(dark ? DARK : LIGHT);
    }

    public static String toggleLabel() {
        return IS_DARK ? TOGGLE_LABEL_LIGHT : TOGGLE_LABEL_DARK;
    }

    private static void apply(Palette palette) {
        COLOR_BG = palette.bg;
        COLOR_PANEL = palette.panel;
        COLOR_SECTION = palette.section;
        COLOR_BORDER = palette.border;
        COLOR_TEXT = palette.text;
        COLOR_MUTED = palette.muted;
        COLOR_MUTED_DIM = palette.mutedDim;
        COLOR_PRIMARY = palette.primary;
        COLOR_PRIMARY_LIGHT = palette.primaryLight;
        COLOR_JSON_BG = palette.jsonBg;
        COLOR_JSON_TEXT = palette.jsonText;
        COLOR_STATUS_BG = palette.statusBg;
        COLOR_STATUS_TEXT = palette.statusText;
        COLOR_STATUS_ERROR_BG = palette.statusErrorBg;
        COLOR_STATUS_ERROR_TEXT = palette.statusErrorText;
        COLOR_DANGER = palette.danger;
        COLOR_TEMPLATE = palette.template;
        GRADIENT_HEADER = palette.headerGradient;
    }

    private static final class Palette {
        private final String bg;
        private final String panel;
        private final String section;
        private final String border;
        private final String text;
        private final String muted;
        private final String mutedDim;
        private final String primary;
        private final String primaryLight;
        private final String jsonBg;
        private final String jsonText;
        private final String statusBg;
        private final String statusText;
        private final String statusErrorBg;
        private final String statusErrorText;
        private final String danger;
        private final String template;
        private final String headerGradient;

        private Palette(String bg,
                        String panel,
                        String section,
                        String border,
                        String text,
                        String muted,
                        String mutedDim,
                        String primary,
                        String primaryLight,
                        String jsonBg,
                        String jsonText,
                        String statusBg,
                        String statusText,
                        String statusErrorBg,
                        String statusErrorText,
                        String danger,
                        String template,
                        String headerGradient) {
            this.bg = bg;
            this.panel = panel;
            this.section = section;
            this.border = border;
            this.text = text;
            this.muted = muted;
            this.mutedDim = mutedDim;
            this.primary = primary;
            this.primaryLight = primaryLight;
            this.jsonBg = jsonBg;
            this.jsonText = jsonText;
            this.statusBg = statusBg;
            this.statusText = statusText;
            this.statusErrorBg = statusErrorBg;
            this.statusErrorText = statusErrorText;
            this.danger = danger;
            this.template = template;
            this.headerGradient = headerGradient;
        }
    }
}
